I attempted a live run rather than leaving the caveat abstract. It did not complete, but it produced three facts worth having, and one of them corrects the prerequisite I wrote in the PR body.

## What worked

Launched with the recipe the sibling probes use, against an isolated profile:

```bash
TUFF_STARTUP_BENCHMARK_ONCE=1 \
TUFF_STARTUP_BENCHMARK_EXIT_DELAY_MS=90000 \
TUFF_STARTUP_BENCHMARK_USER_DATA_DIR=/tmp/tuff-rn-probe \
electron out/main/index.js --remote-debugging-port=9333
```

- **The isolation works exactly as designed.** `[TouchApp] Startup benchmark mode enabled, skip single-instance lock` — so this cannot collide with a running instance, and `precore.ts:160` repoints `userData` at the temp dir. Confirmed in the log, not assumed.
- **CDP comes up in about 5 seconds**, serving one `page` target.

## What did not, and why

The app shut down 4.3 seconds after start, before the probe could attach:

```
[15:04:17.040] [ERROR] [TouchApp] TouchApp runtime initialization failed
                       error=Error: ELECTRON_RENDERER_URL is not set
[15:04:17.632] [ERROR] [CoreBox:Window] Failed to load content in new box window
[15:04:17.635] [DEBUG] [CoreBox:Window] BoxWindow closed
[15:04:20.707] [INFO]  [TouchApp] App quit requested
```

`out/main/index.js` from `electron-vite build` is not self-sufficient — it expects `ELECTRON_RENDERER_URL` pointing at a running Vite dev server. Without it the runtime fails to initialise, the window has nothing to load, and the app cascades to shutdown.

**This corrects the PR body.** I wrote "start CoreApp with `--remote-debugging-port` and an isolated `userDataDir`", which is incomplete. The real prerequisite is one of:

- a **packaged** app — which is why every sibling probe takes an `executablePath` rather than a build directory, a detail I had read past, or
- a running dev server plus `ELECTRON_RENDERER_URL`

I have updated nothing in the branch for this yet; say which shape you want and I will fold it into the header comment, or add the launch half so the probe takes `--app-bundle` like `coreapp-packaged-indexing-diagnostics-probe.ts` does.

## What is still unverified

The two timing assumptions I flagged originally remain untested: that the dialog appears before the router is usable, and that `#/setting/update` renders with the dialog dismissed rather than behind it. Nothing in this attempt got far enough to observe either.

The unit half is unaffected — 16/16 still pass, and none of them mock CDP into passing, which is why they stayed honest about this.

Cleaned up after the attempt: temp profile removed, no stray process.

--- a/apps/core-app/scripts/coreapp-visible-release-notes-probe.test.ts
+++ b/apps/core-app/scripts/coreapp-visible-release-notes-probe.test.ts
@@ -3,8 +3,10 @@ import {
   buildReleaseNotesEvidenceChecks,
   navigateToUpdatePageExpression,
   readReleaseNotesDomExpression,
+  attachableTargets,
+  isDevToolsTarget,
+  isMainWindowIdentity,
   screenshotFileName,
-  selectMainWindowTarget,
   UPDATE_ROUTE_HASH,
   VIEWPORTS,
   type DevToolsTarget,
@@ -87,42 +89,57 @@ describe('buildReleaseNotesEvidenceChecks', () => {
   })
 })
 
-describe('selectMainWindowTarget', () => {
-  const page = (url: string, id = url): DevToolsTarget => ({
+describe('target selection', () => {
+  const page = (id: string, url = 'http://127.0.0.1:9444/'): DevToolsTarget => ({
     id,
     type: 'page',
     url,
-    webSocketDebuggerUrl: `ws://127.0.0.1:9222/devtools/page/${id}`
-  })
-
-  it('returns null when nothing is attachable', () => {
-    expect(selectMainWindowTarget([])).toBeNull()
-    expect(selectMainWindowTarget([{ id: 'a', type: 'service_worker' }])).toBeNull()
-  })
-
-  it('ignores a page target with no websocket url', () => {
-    expect(selectMainWindowTarget([{ id: 'a', type: 'page', url: 'app://index.html' }])).toBeNull()
+    webSocketDebuggerUrl: `ws://127.0.0.1:9333/devtools/page/${id}`
   })
 
   /**
-   * The race this exists to remove: on first launch CoreBox and plugin Surfaces are also `page`
-   * targets, and taking the first one screenshots whichever happened to come up first.
+   * Every case below is a shape a live launch actually produced. A dev-server launch exposed four
+   * page targets: two DevTools windows, the main window and CoreBox — with the main window and
+   * CoreBox reporting the *same* URL, which is why the first version of this probe attached to
+   * CoreBox, reported `#/home` instead of the route it had just set, and captured two screenshots
+   * of the wrong window under the right filename.
    */
-  it('prefers the main window over CoreBox and plugin surfaces', () => {
-    const targets = [
-      page('app://corebox/index.html', 'corebox'),
-      page('http://127.0.0.1:5173/plugin/demo', 'plugin'),
-      page('app://index.html#/', 'main')
+  it('drops DevTools windows, which are page targets and answer Runtime.evaluate happily', () => {
+    expect(isDevToolsTarget({ id: 'a', type: 'page', title: 'DevTools' })).toBe(true)
+    expect(
+      isDevToolsTarget({ id: 'b', type: 'page', url: 'devtools://devtools/bundled/x.html' })
+    ).toBe(true)
+    expect(isDevToolsTarget(page('c'))).toBe(false)
+  })
+
+  it('keeps only attachable non-DevTools pages', () => {
+    const targets: DevToolsTarget[] = [
+      { id: 'devtools', type: 'page', title: 'DevTools', webSocketDebuggerUrl: 'ws://x' },
+      { id: 'no-socket', type: 'page' },
+      { id: 'worker', type: 'service_worker', webSocketDebuggerUrl: 'ws://y' },
+      page('main')
     ]
-    expect(selectMainWindowTarget(targets)?.id).toBe('main')
+    expect(attachableTargets(targets).map((target) => target.id)).toEqual(['main'])
   })
 
-  it('accepts a dev-server main window', () => {
-    expect(selectMainWindowTarget([page('http://localhost:5173/#/', 'dev')])?.id).toBe('dev')
+  it('returns nothing rather than guessing when there is no page target', () => {
+    expect(attachableTargets([])).toEqual([])
   })
 
-  it('falls back to the first page rather than giving up', () => {
-    expect(selectMainWindowTarget([page('app://something-else', 'x')])?.id).toBe('x')
+  /** The discriminator the live run proved necessary: body class, not URL. */
+  it('tells the main window from CoreBox by body class', () => {
+    expect(isMainWindowIdentity({ bodyClass: 'MacIntel', title: 'Tuff' })).toBe(true)
+    expect(isMainWindowIdentity({ bodyClass: 'MacIntel core-box', title: 'Tuff' })).toBe(false)
+    expect(isMainWindowIdentity({ bodyClass: 'plugin-view dark', title: 'Tuff' })).toBe(false)
+  })
+
+  /** `core-boxed` must not be read as `core-box`; the check is on whole class tokens. */
+  it('matches whole class tokens, not substrings', () => {
+    expect(isMainWindowIdentity({ bodyClass: 'core-boxed', title: 'Tuff' })).toBe(true)
+  })
+
+  it('rejects a DevTools identity even if its body class looks ordinary', () => {
+    expect(isMainWindowIdentity({ bodyClass: 'undocked inactive', title: 'DevTools' })).toBe(false)
   })
 })
 

--- a/apps/core-app/scripts/coreapp-visible-release-notes-probe.ts
+++ b/apps/core-app/scripts/coreapp-visible-release-notes-probe.ts
@@ -61,6 +61,19 @@ export const VIEWPORTS: readonly Viewport[] = [
   { name: 'narrow', width: 720, height: 900 }
 ]
 
+/**
+ * Read from a candidate target to decide whether it is the main window.
+ *
+ * Learned by running this: in a dev-server launch the main window and CoreBox load the *same*
+ * base URL, so URL matching cannot tell them apart -- the probe attached to CoreBox, reported
+ * `#/home` instead of the route it had just set, and captured two screenshots of the wrong
+ * window under the right filename. The body class is what actually differs.
+ */
+export interface TargetIdentity {
+  bodyClass: string
+  title: string
+}
+
 /** What one capture pass reads out of the renderer before it shoots. */
 export interface ReleaseNotesProbeDom {
   href: string
@@ -110,19 +123,36 @@ export interface ReleaseNotesEvidenceCheck {
  * several. Picking the first `page` would attach to whichever came up first, which is a race —
  * the probe would sometimes screenshot a plugin's HTML and report a missing dialog.
  */
-export function selectMainWindowTarget(targets: readonly DevToolsTarget[]): DevToolsTarget | null {
-  const pages = targets.filter(
-    (target) => target.type === 'page' && Boolean(target.webSocketDebuggerUrl)
+export function isDevToolsTarget(target: DevToolsTarget): boolean {
+  return (target.url ?? '').startsWith('devtools://') || target.title === 'DevTools'
+}
+
+/**
+ * Attachable, non-DevTools page targets. A live launch exposes one `page` per DevTools window as
+ * well as per app window, and those answer `Runtime.evaluate` perfectly happily — they just report
+ * an empty hash and no dialog, which reads as "the app is broken".
+ */
+export function attachableTargets(targets: readonly DevToolsTarget[]): DevToolsTarget[] {
+  return targets.filter(
+    (target) =>
+      target.type === 'page' && Boolean(target.webSocketDebuggerUrl) && !isDevToolsTarget(target)
   )
-  if (pages.length === 0) return null
-  const main = pages.find((target) => {
-    const url = target.url ?? ''
-    // The main window is the one serving the renderer bundle: `index.html` in a packaged build,
-    // the dev server root in dev. CoreBox and plugin Surfaces carry a route or a plugin origin.
-    if (url.includes('/plugin') || url.includes('corebox') || url.includes('core-box')) return false
-    return url.includes('index.html') || /^https?:\/\/(localhost|127\.0\.0\.1)[:/]/.test(url)
-  })
-  return main ?? pages[0] ?? null
+}
+
+/**
+ * The main window, told apart from CoreBox by body class rather than by URL.
+ *
+ * `core-box` is set on CoreBox's body; the main window carries neither it nor a plugin marker.
+ * URL matching does not work: under a dev-server launch both windows load the same origin.
+ */
+export function isMainWindowIdentity(identity: TargetIdentity): boolean {
+  if (identity.title === 'DevTools') return false
+  const cls = identity.bodyClass
+  return !cls.split(/\s+/).some((token) => token === 'core-box' || token === 'plugin-view')
+}
+
+export function identityExpression(): string {
+  return `JSON.stringify({ bodyClass: document.body?.className || '', title: document.title || '' })`
 }
 
 /**
@@ -224,6 +254,9 @@ export function navigateToUpdatePageExpression(): string {
   })()`
 }
 
+/** Vue Router resolves a hash change on a microtask plus a render tick; this is slack over that. */
+export const ROUTE_SETTLE_MS = 750
+
 export function screenshotFileName(viewport: Viewport, surface: 'dialog' | 'update-page'): string {
   return `release-notes-${surface}-${viewport.name}-${viewport.width}x${viewport.height}.png`
 }
@@ -282,7 +315,13 @@ export async function runProbe(send: CdpSend, outDir: string): Promise<ProbeRunR
   for (const viewport of VIEWPORTS) {
     await applyViewport(send, viewport)
     const dom = await readDom(send)
-    if (viewport.name === 'desktop') checks.push(...buildReleaseNotesEvidenceChecks(dom))
+    if (viewport.name === 'desktop') {
+      checks.push(
+        ...buildReleaseNotesEvidenceChecks(dom).filter(
+          (check) => check.tag !== 'update-page-reached'
+        )
+      )
+    }
     if (dom.hasDialog) {
       const file = path.join(outDir, screenshotFileName(viewport, 'dialog'))
       await capture(send, file)
@@ -294,12 +333,24 @@ export async function runProbe(send: CdpSend, outDir: string): Promise<ProbeRunR
     expression: navigateToUpdatePageExpression(),
     returnByValue: true
   })
+  // The router resolves the hash asynchronously; capturing immediately shoots the previous route.
+  await new Promise((resolve) => setTimeout(resolve, ROUTE_SETTLE_MS))
 
-  for (const viewport of VIEWPORTS) {
-    await applyViewport(send, viewport)
-    const file = path.join(outDir, screenshotFileName(viewport, 'update-page'))
-    await capture(send, file)
-    artifacts.push(file)
+  const afterNav = await readDom(send)
+  const navChecks = buildReleaseNotesEvidenceChecks(afterNav).filter(
+    (check) => check.tag === 'update-page-reached'
+  )
+  checks.push(...navChecks)
+
+  // Only capture what the checks say is actually on screen. A PNG named `update-page` that shows
+  // something else is worse than no PNG: it is filed as evidence and read as one.
+  if (navChecks.every((check) => check.ok)) {
+    for (const viewport of VIEWPORTS) {
+      await applyViewport(send, viewport)
+      const file = path.join(outDir, screenshotFileName(viewport, 'update-page'))
+      await capture(send, file)
+      artifacts.push(file)
+    }
   }
 
   return { checks, artifacts }
@@ -310,7 +361,9 @@ function arg(name: string): string | undefined {
   return index >= 0 ? process.argv[index + 1] : undefined
 }
 
-function isDevToolsTarget(value: unknown): value is DevToolsTarget {
+/** Shape guard for the `/json` payload -- distinct from `isDevToolsTarget`, which asks whether a
+ * target *is* a DevTools window. The two were briefly the same name, which esbuild caught. */
+function hasDevToolsTargetShape(value: unknown): value is DevToolsTarget {
   if (!value || typeof value !== 'object') return false
   const target = value as Partial<DevToolsTarget>
   return typeof target.id === 'string' && typeof target.type === 'string'
@@ -322,7 +375,7 @@ async function loadTargets(remoteDebuggingUrl: string): Promise<DevToolsTarget[]
   const payload = (await response.json()) as unknown
   if (!Array.isArray(payload))
     throw new Error('Remote debugging endpoint did not return a target list')
-  return payload.filter(isDevToolsTarget)
+  return payload.filter(hasDevToolsTargetShape)
 }
 
 async function withTarget<T>(
@@ -387,9 +440,34 @@ async function main(): Promise<void> {
   }
 
   const targets = await loadTargets(`${cdpUrl.replace(/\/$/, '')}/json`)
-  const target = selectMainWindowTarget(targets)
-  if (!target) {
+  const candidates = attachableTargets(targets)
+  if (candidates.length === 0) {
     console.error(`no attachable page target among ${targets.length} target(s)`)
+    process.exit(1)
+  }
+
+  // Identity has to be read from the renderer, not guessed from the target list: under a
+  // dev-server launch every window reports the same URL.
+  let target: DevToolsTarget | null = null
+  for (const candidate of candidates) {
+    const identity = await withTarget(candidate, async (send) => {
+      const response = await send('Runtime.evaluate', {
+        expression: identityExpression(),
+        returnByValue: true
+      })
+      const raw = (response.result?.result as { value?: string } | undefined)?.value
+      return raw ? (JSON.parse(raw) as TargetIdentity) : null
+    })
+    if (identity && isMainWindowIdentity(identity)) {
+      target = candidate
+      break
+    }
+  }
+  if (!target) {
+    console.error(
+      `none of ${candidates.length} attachable target(s) is the main window ` +
+        '(all were DevTools, CoreBox or plugin views)'
+    )
     process.exit(1)
   }
 

--- a/apps/core-app/scripts/coreapp-visible-release-notes-probe.ts
+++ b/apps/core-app/scripts/coreapp-visible-release-notes-probe.ts
@@ -5,11 +5,25 @@
  * Surfaces, and why each is reachable:
  *
  * - `WhatsChangedDialog` — rendered by `AppEntrance.vue` when `appEntranceMode === 'MainApp'`.
- *   `useReleaseNotesRuntime.evaluateStartup()` opens it only while the version is unacknowledged,
- *   and `closeDialog()` acknowledges it permanently. So it appears exactly once per profile per
- *   upgrade: **this probe is only reproducible against a fresh `userDataDir`**, which is also why
- *   every sibling probe isolates one. Pointed at a real profile it captures nothing and reports
- *   `dialog-not-shown`, which is a true observation about that profile, not a probe failure.
+ *   Capturing it needs **three** conditions, all learned by running this and finding it absent:
+ *
+ *     1. The bundled catalog must resolve. `release-notes-service.ts:readCatalog` walks
+ *        `catalogPaths` and throws `Bundled release notes catalog is missing` when every candidate
+ *        is ENOENT. Under a dev-build launch (`electron out/main/index.js`) none of them reach the
+ *        repo's `resources/release-notes/catalog.json`, so `getBundledReleaseNotes()` fails and
+ *        `evaluateStartup` never reaches a decision. **A packaged app is required**, not merely
+ *        convenient.
+ *     2. Onboarding must be complete. `evaluateStartup(Boolean(appSetting?.beginner?.init))` feeds
+ *        `resolveReleaseNotesStartupDecision`, whose first branch is
+ *        `if (!onboardingComplete) return { kind: 'acknowledge' }` — it marks the version seen and
+ *        shows nothing. A brand-new profile has `beginner.init === false`, verified on disk.
+ *     3. The version must be unacknowledged, and the catalog must contain an entry matching its own
+ *        `generatedForVersion`; otherwise the same `acknowledge` branch is taken.
+ *
+ *   So an isolated profile is necessary but **not sufficient** — and on its own it actively
+ *   guarantees the dialog will not appear, because (2) fails and silently acknowledges. Seed
+ *   `<userDataDir>/tuff-dev/modules/config/app-setting.ini` with `beginner.init: true` before the
+ *   first launch.
  * - Settings → Update (`router.ts` key `update` → `SettingUpdatePage.vue`). #482 and the
  *   maintenance audit both call this "Update-history"; no such name exists in the renderer —
  *   searching all 756 files for `update.?history` returns zero. The name here follows the router.

--- a/apps/core-app/scripts/coreapp-visible-release-notes-probe.ts
+++ b/apps/core-app/scripts/coreapp-visible-release-notes-probe.ts
@@ -7,12 +7,14 @@
  * - `WhatsChangedDialog` — rendered by `AppEntrance.vue` when `appEntranceMode === 'MainApp'`.
  *   Capturing it needs **three** conditions, all learned by running this and finding it absent:
  *
- *     1. The bundled catalog must resolve. `release-notes-service.ts:readCatalog` walks
- *        `catalogPaths` and throws `Bundled release notes catalog is missing` when every candidate
- *        is ENOENT. Under a dev-build launch (`electron out/main/index.js`) none of them reach the
- *        repo's `resources/release-notes/catalog.json`, so `getBundledReleaseNotes()` fails and
- *        `evaluateStartup` never reaches a decision. **A packaged app is required**, not merely
- *        convenient.
+ *     1. The bundled catalog must resolve, and in a dev build that depends on the **working
+ *        directory**. `UpdateService.getReleaseNotesCatalogPaths()`'s last candidate is
+ *        `process.cwd()/apps/core-app/resources/release-notes/catalog.json`, so launching from the
+ *        repo root works and launching from `apps/core-app` does not -- the path doubles to
+ *        `apps/core-app/apps/core-app/...`. When every candidate misses,
+ *        `readCatalog` throws `Bundled release notes catalog is missing`,
+ *        `getBundledReleaseNotes()` fails, and `evaluateStartup` never reaches a decision.
+ *        (I first read this as "a packaged app is required". It is not; the cwd was wrong.)
  *     2. Onboarding must be complete. `evaluateStartup(Boolean(appSetting?.beginner?.init))` feeds
  *        `resolveReleaseNotesStartupDecision`, whose first branch is
  *        `if (!onboardingComplete) return { kind: 'acknowledge' }` — it marks the version seen and
@@ -33,8 +35,15 @@
  * is exercised only by a real run.
  *
  * Launching is left to the caller, the same split `coreapp-packaged-indexing-diagnostics-probe.ts`
- * already implements: start CoreApp with `--remote-debugging-port` and an isolated `userDataDir`,
- * then point this at it.
+ * already implements. A dev-build launch that satisfies all three conditions, verified end to end:
+ *
+ *   (cd apps/core-app/out/renderer && python3 -m http.server 9447 &)
+ *   mkdir -p /tmp/p/tuff-dev/modules/config
+ *   echo '{"beginner":{"init":true}}' > /tmp/p/tuff-dev/modules/config/app-setting.ini
+ *   # from the REPO ROOT, not apps/core-app -- see condition 1
+ *   ELECTRON_RENDERER_URL=http://127.0.0.1:9447/ TUFF_STARTUP_BENCHMARK_ONCE=1 \
+ *     TUFF_STARTUP_BENCHMARK_EXIT_DELAY_MS=60000 TUFF_STARTUP_BENCHMARK_USER_DATA_DIR=/tmp/p \
+ *     electron apps/core-app/out/main/index.js --remote-debugging-port=9333
  *
  *   tsx scripts/coreapp-visible-release-notes-probe.ts \
  *     --cdp-url http://127.0.0.1:9222 --out-dir docs/engineering/reports/release-notes-<date>

